packages/vite/src/node/utils.ts

```diff
@@ -825,6 +825,10 @@ export interface Hostname {
 export async function resolveHostname(
   optionsHost: string | boolean | undefined
 ): Promise<Hostname> {
+  // If passed multiple redundant --host, use the first one
+  if (Array.isArray(optionsHost)) {
+    optionsHost = optionsHost[0]
+  }
```

Comment:
I suppose we want to sanitize this early on at https://github.com/vitejs/vite/blob/c4489ea193f4d2f703882cb03490980aa5174040/packages/vite/src/node/cli.ts#L78

but it looks like technically all options can become an array and we'll have to sanitize all of them 🤔

Reply:
ok. i will sanitize all options.

Reply:
I don't think this solution is very elegant.